tests/test_github_requirements.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python
+
+# Copyright (c) 2016 IBM Corp.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import logging
+
+from tests.base import ZuulTestCase
+
+logging.basicConfig(level=logging.DEBUG,
+                    format='%(asctime)s %(name)-32s '
+                    '%(levelname)-8s %(message)s')
+
+
+class TestGithubRequirements(ZuulTestCase):
+    """Test pipeline and trigger requirements"""
+
+    def setup_config(self, config_file='zuul-github.conf'):
+        super(TestGithubRequirements, self).setup_config(config_file)
+
+    def test_pipeline_require_status(self):
+        "Test pipeline requirement: status"
+        return self._test_require_status('org/project1',
+                                         'project1-pipeline')
+
+#    def test_trigger_require_status(self):
+#        "Test trigger requirement: status"
+#        return self._test_require_status('org/project2',
+#                                         'project2-trigger')
+
+    def _test_require_status(self, project, job):
+        self.config.set('zuul', 'layout_config',
+                        'tests/fixtures/layout-github-requirement-status.yaml')
+        self.sched.reconfigure(self.config)
+        self.registerJobs()
+
+        A = self.fake_github.openFakePullRequest(project, 'master', 'A')
+        # A comment event that we will keep submitting to trigger
+        comment = A.getCommentAddedEvent('test me')
+        self.fake_github.emitEvent(comment)
+        self.waitUntilSettled()
+        # No status from zuul so should not be enqueued
+        self.assertEqual(len(self.history), 0)
+
+        # An error status should not cause it to be enqueued
+        A.setStatus(A.head_sha, 'error', 'null', 'null', 'check')
+        self.fake_github.emitEvent(comment)
+        self.waitUntilSettled()
+        self.assertEqual(len(self.history), 0)
+
+        # A success status goes in
+        A.setStatus(A.head_sha, 'success', 'null', 'null', 'check')
+        self.fake_github.emitEvent(comment)
+        self.waitUntilSettled()
+        self.assertEqual(len(self.history), 1)
+        self.assertEqual(self.history[0].name, job)
+
+# TODO: Implement reject on status
```

Comment:
I decided to leave it as a TODO for a later time, if we decide we want to do that.

I'm not really sure we want to, I don't think it makes much sense. Maybe we could do it with PR reviews, as PR reviews more closely align with approvals, so it wouldn't take as much work.

Once this PR goes in, I was going to work on requirement for PR reviews as approvals.